cmd/wharf/provisioner.go

```diff
@@ -32,8 +32,18 @@ orchestration is handled inside the Kubernetes cluster, in comparison to the
 		if err != nil {
 			return err
 		}
+		// Question: Should the flag override the wharf-cmd-config.yml,
+		// vice-versa, or one of them act as a fallback?
+		//
+		// Currently going with value from provisionerFlags as fallback.
 		provisionerFlags.restConfig = restConfig
 		provisionerFlags.namespace = ns
+		if rootConfig.Provisioner.K8s.Namespace == "default" && ns != "" {
+			rootConfig.Provisioner.K8s.Namespace = ns
+		}
+		if rootConfig.ProvisionerAPI.K8s.Namespace == "default" && ns != "" {
+			rootConfig.ProvisionerAPI.K8s.Namespace = ns
+		}
 		return nil
```

Comment:
Leaving comment to highlight question

Reply:
Flags has precedence over configs.

Namespace is a little special, and should be overridden in the order of something like:

- value from `~/.kube/config`
- value from `wharf-cmd-config.yml`
- value from `--k8s-namespace`

You could set the provisioner flag's defaults from the configs to achive this.